src/Microsoft.Diagnostics.Runtime/Utilities/DumpReader.cs

```diff
@@ -2522,4 +2721,350 @@ internal enum ProcessorArchitecture : ushort
+        /// </summary>
+        public uint Attributes { get; private set; }
+        /// <summary>
+        /// The meaning of this member depends on the handle type and the operating system.
+        /// </summary>
+        public uint GrantedAccess { get; private set; }
+        /// <summary>
+        ///  An RVA to a MINIDUMP_HANDLE_OBJECT_INFORMATION sttructure
+        /// </summary>
+        public Int32 ObjectInfoRva { get; private set; }
+        /// <summary>
+        /// Type of MiniDump handle
+        /// </summary>
+        public DumpHandleType Type { get; set; }
+        /// <summary>
+        /// </summary>
+        public uint OwnerProcessId { get; internal set; }
+        /// <summary>
+        /// Handle owner thread id
+        /// </summary>
+        public uint OwnerThreadId { get; internal set; }
+        /// <summary>
+        /// Unknown mutex fields
+        /// </summary>
+        public MutexUnknownFields MutexUnknown { get; internal set; }
+
+        /// <summary>
+        /// Additional minidump handle
+        /// </summary>
+        public List<DumpHandleInfo> HandleInfoList { get; private set; }
+
+        internal void AddInfo(MINIDUMP_HANDLE_OBJECT_INFORMATION info)
+        {
+            if (HandleInfoList == null)
+            {
+                HandleInfoList = new List<DumpHandleInfo>();
+            }
+            HandleInfoList.Add(new DumpHandleInfo(info));
+        }
+    }
+
+    /// <summary>
+    /// MiniDumpHandle type enumeration
+    /// </summary>
+    public enum DumpHandleType
+    {
+        /// <summary>
+        /// No Type
+        /// </summary>
+        NONE,
+        /// <summary>
+        /// Thread Type
+        /// </summary>
+        THREAD,
+        /// <summary>
+        /// Mutex Type
+        /// </summary>
+        MUTEX1,
+        /// <summary>
+        /// Mutex Type
+        /// </summary>
+        MUTEX2,
+        /// <summary>
+        /// Process Type
+        /// </summary>
+        PROCESS1,
+        /// <summary>
+        /// Process Type
+        /// </summary>
+        PROCESS2,
+        /// <summary>
+        /// Event Type
+        /// </summary>
+        EVENT,
+        /// <summary>
+        /// Critical Section Type
+        /// </summary>
+        SECTION,
+        /// <summary>
+        /// Max Type
+        /// </summary>
+        TYPE_MAX
+    }
+    /// <summary>
+    /// Mutex unknown fields
+    /// </summary>
+    public class MutexUnknownFields
+    {
+        /// <summary>
+        /// First mutex unknown field
+        /// </summary>
+        public object Field1 { get; internal set; }
+        /// <summary>
+        /// Second mutex unknown field
+        /// </summary>
+        public object Field2 { get; internal set; }
+    }
+
+    /// <summary>
+    /// Handle additional object info constructed by MINIDUMP_HANDLE_OBJECT_INFORMATION structure 
+    /// </summary>
+    public class DumpHandleInfo
```

Comment:
How are clients expected to use this? Are they supposed to look at `InfoType` and then decide which of the following internal structures to read? If so, there should be a pointer somewhere in this class pointing to where that structure is in the dump.